python/tests/unit/arcticdb/version_store/test_parallel.py

```diff
@@ -240,3 +242,48 @@ def test_datetimes_to_nats(lmdb_version_store):
         result = result.astype(df.dtypes)
 
     assert_frame_equal(result, df)
+
+
+@pytest.mark.parametrize("append", (True, False))
+@pytest.mark.parametrize("arg", (True, False, None))
+@pytest.mark.parametrize("lib_config", (True, False))
+def test_compact_incomplete_prune_previous(lib_config, arg, append, version_store_factory):
+    lib = version_store_factory(prune_previous_version=lib_config)
+    lib.write("sym", pd.DataFrame({"col": [3]}, index=pd.DatetimeIndex([0])))
+    lib.append("sym", pd.DataFrame({"col": [4]}, index=pd.DatetimeIndex([1])), incomplete=True)
+
+    lib.compact_incomplete("sym", append, convert_int_to_float=False, prune_previous_version=arg)
+    assert lib.read_metadata("sym").version == 1
+
+    should_prune = lib_config if arg is None else arg
+    assert lib.has_symbol("sym", 0) == (not should_prune)
```

Comment:
Not your test I know but why isn't this `!= should_prune` :(

Reply:
No idea, will change

Reply:
I accidentally committed to this branch for a PR that's gonna follow, so this has gone